File: binance4j-websocket/src/test/java/com/binance4j/client/WebsocketTradeClientTest.java
package com.binance4j.client;

import java.util.concurrent.ExecutionException;

import org.junit.jupiter.api.Test;

import com.binance4j.websocket.client.WebsocketTradeClient;
import com.binance4j.websocket.dto.Trade;

class WebsocketTradeClientTest {
	TestCallback<Trade> callback = new TestCallback<>();

	@Test
	void test1() throws InterruptedException, ExecutionException {
		WebsocketTradeClient client = new WebsocketTradeClient(callback.getSymbol(), callback);
		callback.setWebsocketClient(client);
		client.open();
		callback.future.get();
	}
}

File: connectors/src/test/test/com/binance4j/websocket/WebsocketTradeClientTest.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package com.binance4j.websocket

import com.binance4j.websocket.client.WebsocketTradeClient
import com.binance4j.websocket.dto.Trade
import org.junit.jupiter.api.Test

internal class WebsocketTradeClientTest {
	var callback = TestCallback<Trade>()
	
	@Test
	fun test1() {
		val client = WebsocketTradeClient(callback.symbol, callback)
		callback.websocketClient = client
		client.open()
		callback.future.get()
	}
}